pkg/lib/machine-info.es6

```diff
@@ -191,3 +191,21 @@ export function udev_info(address) {
     }
     return pr;
 }
+
+var dmide_info_promises = { };
+
+export function dmide_info(address) {
+    var pr = dmide_info_promises[address];
+    var dfd;
+
+    if (!pr) {
+        dfd = cockpit.defer();
+        dmide_info_promises[address] = pr = dfd.promise();
+
+        cockpit.spawn(["/usr/sbin/dmidecode", "-t", "17"], { err: "message" })
+            .done(output => dfd.resolve(output))
```

Comment:
Let's do the structural parsing of the raw text into a dictionary here already (without interpreting, filtering, or renaming keys), similar to `parseUdevDB()`. This makes the interface more abstract, and callers can then pick out the info they want.